src/diagnostics/land_compute_methods.jl

```diff
@@ -61,88 +61,130 @@ end
 @diagnostic_compute "subsurface_water_storage" BucketModel Y.bucket.W
 @diagnostic_compute "surface_water_content" BucketModel Y.bucket.Ws
 
-### SoilCanopyModel ###
+### Union{SoilCanopyModel, LandModel} ###
```

Comment:
Seeing that we have so many instances of `Union{SoilCanopyModel, LandModel}` here, it might be good to update our type hierarchy with a new abstract type - something like `AbstractSoilCanopyModel`, where we would have both `SoilCanopyModel <: AbstractSoilCanopyModel` and  `LandModel <: AbstractSoilCanopyModel`. Maybe even `AbstractIntegratedModel` or `AbstractComplexModel`, but these values are specific to having canopy + soil so we might want to be more specific in the naming.

We don't have to do this in this PR, just something to think about

Reply:
Why not `AbstractLandModel` for the parent type? (we can discuss for another PR)

Reply:
^^ I like this direction of thinking! Right now "AbstractLandModel <: AbstractModel" is taken (it is the abstract type of SoilCanopyModel, LandModel, LandHydrologyModel, etc). Like Julia wrote, some of these methods only make sense if we have a canopy, so we cant use it as is. Ill open an issue!

Reply:
Issue #895